ubuntu-advantage-client/ec2_cleanup.py

```diff
@@ -19,62 +19,125 @@ def parse_args():
         help=("Tag used to filter cloud resources for deletion. "
               "Default: {}".format(CI_DEFAULT_TAG))
     )
+    parser.add_argument(
+        "-o", "--older-than", dest="older_than", action="store",
+        help=("Tag used to filter cloud resources for deletion. "
+              "Format: MM/DD/YY [HH:MM:SS].)"
+    )
     return parser.parse_args()
 
 
-def clean_ec2(tag):
+def get_tag_prefix(older_than):
+    if not older_than:
+       return ""
+    print('Match only resources created before %s' % older_than)
+    try:
+        time = datetime.datetime.strptime(older_than, "%m/%d/%y %H:%M:%S")
+        return time.strftime("uaclient-ci-%m%d-%H%M%S")
+    except ValueError:
+        time = datetime.datetime.strptime(older_than, "%m/%d/%y")
+        return time.strftime("uaclient-ci-%m%d")
+
+
+def is_resource_stale(resource, tag_time_prefix):
+    """Return True if the given resource is older than the tag_time_prefix.
+
+    If no Name tag present, assume stale and return True
+    """
+    for tag in resource.tags:
+        if tag["Key"] != "Name":
+            continue
+        if tag["Value"] >= tag_time_prefix:
+            return False
+    return True
+
+
+def clean_ec2(tag_prefix, older_than=None):
     """Clean up all running EC2 instances, VPCs, and storage."""
     client = boto3.client('ec2')
     resource = boto3.resource('ec2')
 
-    print('# searching for vpcs matching tag {}'.format(tag))
-    tag_filter = [{'Name': 'tag:Name', 'Values': [tag]}]
-    for vpc in list(resource.vpcs.filter(Filters=tag_filter)):
+    tag_time_prefix = get_tag_prefix(older_than)
```

Comment:
Shouldn't the `tag_time_prefix` be constructed regarding the `tag` parameter ?

Reply:
Yes it should . just pushed a change for this.